src/lib.rs

```diff
@@ -29,27 +29,62 @@ pub use crate::node::Taffy;
 #[cfg(feature = "std")]
 use core::fmt::{Display, Formatter, Result};
 
-/// An error that can occur when performing layout
+/// The [`Node`](node::Node) was not found in the [`Taffy`] instance
 #[derive(Debug)]
-pub enum Error {
-    /// The [`Node`](node::Node) was invalid
-    InvalidNode(node::Node),
+pub struct NodeNotFoundError(pub node::Node);
+
+#[cfg(feature = "std")]
+impl Display for NodeNotFoundError {
+    fn fmt(&self, f: &mut Formatter) -> Result {
+        write!(f, "Node {:?} is not in the Taffy instance", self.0)
+    }
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for NodeNotFoundError {}
+
+/// An error that occurs while trying to access or modify a [`Node`](node::Node)'s children by index.
+#[derive(Debug)]
+pub enum ChildOperationError {
```

Comment:
```suggestion
pub enum NodeError {
```

I think that a single unified "errors that occur when working with Nodes" is a solid straightforward design here.